frontend/src/components/transactions/TransactionAction.tsx

```diff
@@ -13,39 +14,80 @@ export interface Props {
   viewMode?: ViewMode;
 }
 
-export default class extends React.Component<Props> {
+interface State {
+  status?: T.ExecutionStatus;
+  isFinal?: boolean;
+}
+
+export default class extends React.Component<Props, State> {
   static defaultProps = {
     viewMode: "sparse",
   };
 
+  state: State = {};
+
+  componentDidMount() {
+    this.fetchFinalStamp();
+    this.fetchStatus();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (
+      this.props.transaction.blockTimestamp !==
+      prevProps.transaction.blockTimestamp
+    ) {
+      this.fetchFinalStamp();
+    }
+  }
+
+  fetchStatus = async () => {
+    const status = await new TransactionsApi().getTransactionStatus(
+      this.props.transaction
+    );
+    this.setState({ status });
+  };
+
+  fetchFinalStamp = async () => {
+    const finalStamp = await new TransactionsApi().queryFinalTimestamp();
+    this.setState({
+      isFinal: this.props.transaction.blockTimestamp <= finalStamp,
+    });
+  };
```

Comment:
Let's use React Context on the App component (pages/_app.jsx) in order to avoid fetching the final timestamp on every single TransactonAction. Final timestamp and block height is useful throughout the Explorer, so it seems that the App component is a reasonable place to query and provide them via Context to all the components below.

Reply:
block height is not in this repo, I will fix the final timestamp first and after merge this, will add block height for another PR.